src/engine/GraphStoreProtocol.h

```diff
@@ -94,43 +105,108 @@ class GraphStoreProtocol {
                      ad_utility::truncateOperationString(rawRequest.body()));
     return res;
   }
-  FRIEND_TEST(GraphStoreProtocolTest, transformPost);
+  FRIEND_TEST(GraphStoreProtocolTest, transformPostAndTsop);
+
+  // Transform a SPARQL Graph Store Protocol TSOP to an equivalent ParsedQuery
+  // which is an SPARQL Update.
+  CPP_template_2(typename RequestT)(
+      requires ad_utility::httpUtils::HttpRequest<RequestT>) static ParsedQuery
+      transformTsop(const RequestT& rawRequest, const GraphOrDefault& graph) {
+    if (rawRequest.body().empty()) {
+      throw HttpError(boost::beast::http::status::no_content);
+    }
```

Comment:
You have this at least 2 times, 

move it to a small helper which also uses a nice message.